src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/branch/BitbucketServerBranch.java

```diff
@@ -80,4 +87,25 @@ public void setTimestamp(long timestamp) {
         this.timestamp = timestamp;
     }
 
+    @Restricted(NoExternalUse.class)
+    public void setTimestampClosure(Callable<Long> timestampClosure) {
+        this.timestampClosure = timestampClosure;
+    }
+
+    private long timestamp() {
```

Comment:
Not thread safe. Do you need that?

Reply:
no need for thread-safe... worst case it should be idempotent